rust-connectors/sinks/kafka/src/main.rs

```diff
@@ -18,15 +26,54 @@ async fn main() -> anyhow::Result<()> {
         println!("{}", serde_json::to_string_pretty(&schema).unwrap());
         return Ok(());
     }
-    let opts: KafkaOpt = KafkaOpt::from_args();
-    opts.common.enable_logging();
+    let raw_opts = KafkaOpt::from_args();
+    raw_opts.common.enable_logging();
     info!(
         connector_version = env!("CARGO_PKG_VERSION"),
         git_hash = git_hash_version(),
         "Starting Kafka sink connector",
     );
 
-    opts.execute().await?;
+    // =======================================================
+    // Setup objects and actors based on opts
+    // =======================================================
+
+    // KafkaSinkWorker members need to be inside Arc because it is referenced inside tokio::spawn
+    // That expect things moved inside to be static
+    // An optimization point for this would be to own the runtime (rather than calling tokio::*)
+    // So that runtime::spawn does not need producer to be in Arc
+    let kafka_sink_deps = KafkaSinkDependencies::from_kafka_opt(raw_opts).await?;
+
+    // =======================================================
+    // Streaming sequence starts here
+    // =======================================================
+
+    info!("Starting stream");
+
+    let mut stream = kafka_sink_deps
+        .common_connector_opt
+        .create_consumer_stream()
+        .await?;
+
+    while let Some(Ok(record)) = stream.next().await {
+        // Clone arcs to be sent to async block's scopes
+        let producer = kafka_sink_deps.kafka_producer.clone();
+        let kafka_partition = kafka_sink_deps.kafka_partition.clone();
+        let kafka_topic = kafka_sink_deps.kafka_topic.clone();
+
+        tokio::spawn(async move {
+            let mut kafka_record = FutureRecord::to(kafka_topic.as_str())
+                .payload(record.value())
+                .key(record.key().unwrap_or(&[]));
+            if let Some(kafka_partition) = &(*kafka_partition) {
+                kafka_record = kafka_record.partition(*kafka_partition);
+            }
+            let future = producer.send(kafka_record, Duration::from_secs(0));
+            let res = future.await;
+            debug!("Kafka produce {:?}", res);
```

Comment:
I think we should do something like this (even though it's a little out of the scope of this PR):
```suggestion
            if let Err(e) = producer.send(kafka_record, Duration::from_secs(0)).await {
                error!("Kafka produce {:?}", e)
            }
```
I shouldn't have left the `debug!` line in when I originally build the connector. The debug output is pretty useless.